src/Mvc/Mvc.ViewFeatures/test/Infrastructure/TempDataSerializerTestBase.cs

```diff
@@ -173,6 +173,72 @@ public void RoundTripTest_GuidValue()
             Assert.Equal(value, roundTripValue);
         }
 
+        [Fact]
+        public virtual void RoundTripTest_DateTimeToString()
+        {
+            // Documents the behavior of round-tripping a DateTime value as a string
+            // Arrange
+            var key = "test-key";
+            var testProvider = GetTempDataSerializer();
+            var value = new DateTime(2009, 1, 1, 12, 37, 43);
+            var input = new Dictionary<string, object>
+            {
+                { key, value.ToString() }
+            };
+
+            // Act
+            var bytes = testProvider.Serialize(input);
+            var values = testProvider.Deserialize(bytes);
+
+            // Assert
+            var roundTripValue = Assert.IsType<string>(values[key]);
+            Assert.Equal(value.ToString(), roundTripValue);
+        }
+
+        [Fact]
+        public virtual void RoundTripTest_StringThatIsNotCompliantGuid()
```

Comment:
Just to verify, we are ok with this, correct?

Reply:
Yup, this is fine. We just need to able to document "a" behavior so we don't find it confusing \ surprising.